components/Spinner.js

```diff
@@ -1,3 +1,96 @@
-// This is a mock files
-// API: you can get a spinner from the UI using js calls for dom manipulation
-// Then you can
+import React from 'react';
+import { Accelerometer, Gyroscope } from 'Sensors';
+import { Text, View, Header } from 'react-native';
+
+const Spinner = React.createClass({
+  propTypes: {
+    triggerMatch: React.propTypes.func,
+    beginMatching: React.propTypes.func,
+    mode: React.propTypes.string
+  },
+
+  getInitialState: function () {
+    return ({
+      accel: {
+        x: 0,
+        y: 0,
+        z: 0
+      },
+      gyro: {
+        x: 0,
+        y: 0,
+        z: 0
+      },
+      matchLevel: 0
+    });
+  },
+
+  componentDidMount: function () {
+    DeviceEventEmitter.addListener('AccelerationData', function(data) {
+      this.setState({
+        accel: {
+          x: data.acceleration.x.toFixed(100),
+          y: data.acceleration.y.toFixed(100),
+          z: data.acceleration.z.toFixed(100)
+        }
+      });
+    }.bind(this));
+
+    DeviceEventEmitter.addListener('GyroscopeData', function(data) {
+      this.setState({
+        gyro: {
+          x: data.rotationRate.x.toFixed(100),
+          y: data.rotationRate.y.toFixed(100),
+          z: data.rotationRate.z.toFixed(100)
+        }
+      });
+    }.bind(this));
+
+    Accelerometer.startAccelerometer();
+    Gyroscope.startGyroUpdates();
+  },
+
+  componentWillUnmount: function () {
+    Gyroscope.stopGyroUpdates();
+    Accelerometer.stopAccelerometer();
+  },
+
+  componentWillUpdate: function (prevState) {
+    { accel, gyro, matchLevel } = this.state;
+    { thresholds, triggerMatch, mode } = this.props;
+
+    const newmatchLevel = Math.abs(gyro.x*gyro.y*gyro.z)/100;
+    const accelLevel = Math.abs(accel.x*accel.y*accel.z);
+
+    if (mode === 'initial' && accelLevel > 100 || newmatchLevel > 10) {
+      this.beginMatching();
+      return;
+    }
+
+    if (accelLevel > 9000) {
```

Comment:
While this seems obvious to me it may not be clear why we picked 9k for everyone. Could you add comment explaining this cut-off?